### PR TITLE
No free lunch forp pluq

### LINKED ISSUE #236: Force parallel PLUQ to use no more than the number of threads provided
re-enable the `threads_fgemm` and `threads_ftrsm` routines and validate by experiments (should not degrade from the current perfs)

### PR DESCRIPTION
**Pull request by ClementPernet**
Fixes #236.
Depends on #241 (the branch of #241 is merged in this one, so either first review #241, which will make this PR simpler to review, or review just this one).

### CODE REVIEW

File: fflas-ffpack/ffpack/ffpack_permutation.inl [hunk 1]
```diff
@@ -357,12 +357,35 @@ namespace FFPACK {
...
+                              const size_t R1, const size_t R2,
+                              const size_t R3, const size_t R4,
+                              const FFLAS::ParSeqHelper::Sequential seq)
     {
         typename Field::Element_ptr tmp = FFLAS::fflas_new (F, M2-R1-R2, width);
         doApplyS (F, A, lda, tmp, width, M2, R1, R2, R3, R4);
         FFLAS::fflas_delete (tmp);
     }
+    template <class Field, class Cut, class Param>
+    void MatrixApplyS (const Field& F, typename Field::Element_ptr A, const size_t lda,
```
Comment (Breush) [inline fflas-ffpack/ffpack/ffpack_permutation.inl]: No `inline`?

File: fflas-ffpack/ffpack/ffpack_permutation.inl [hunk 2]
```diff
@@ -396,12 +419,37 @@ namespace FFPACK {
...
+                              const size_t R3, const size_t R4,
+                              const FFLAS::ParSeqHelper::Sequential seq)
     {
         typename Field::Element_ptr tmp = FFLAS::fflas_new (F, N2-R1, width);
         doApplyT (F, A, lda, tmp, width, N2, R1, R2, R3, R4);
         FFLAS::fflas_delete (tmp);
     }
+
+    template <class Field, class Cut, class Param>
+    void MatrixApplyT (const Field& F, typename Field::Element_ptr A, const size_t lda,
```
Comment (Breush) [inline fflas-ffpack/ffpack/ffpack_permutation.inl]: No `inline`?

File: fflas-ffpack/ffpack/ffpack_permutation.inl [hunk 3]
```diff
@@ -720,103 +768,33 @@ namespace FFPACK {
...
-                        const size_t R1, const size_t R2,
-                        const size_t R3, const size_t R4)
+    //#if defined(__FFLASFFPACK_USE_OPENMP) and defined(_OPENMP)
+    template<class Field, class Cut, class Param>
+    inline void applyP( const Field& F,
+                        const FFLAS::FFLAS_SIDE Side,
+                        const FFLAS::FFLAS_TRANSPOSE Trans,
+                        const size_t m, const size_t ibeg, const size_t iend,
+                        typename Field::Element_ptr A, const size_t lda, const size_t * P,
+                        const FFLAS::ParSeqHelper::Parallel<Cut, Param> PSH)//FFLAS::CuttingStrategy::Block,FFLAS::StrategyParameter::Threads> PSH)
```
Comment (Breush) [inline fflas-ffpack/ffpack/ffpack_permutation.inl]: Leftover comment

**Comment by ClementPernet** [comment]
Review addressed. Merging

**Pull request merged**